src/PowerShellEditorServices/Session/RemoteFileManager.cs

```diff
@@ -36,55 +36,161 @@ public class RemoteFileManager
+                    $container += $Value
+                }
+
+                end {
+                    foreach ($fileName in $Path)
                     {
-                        $params['Encoding']='Byte'
-                    }
+                        if (-not (Test-Path $fileName) -or $Force) {
+                            $container > $fileName
 
-                    $contentBytes = Get-Content @params
+                            # Get file contents
+                            $params = @{ Path=$fileName; Raw=$true }
+                            if ($PSVersionTable.PSEdition -eq 'Core')
+                            {
+                                $params['AsByteStream']=$true
+                            }
+                            else
+                            {
+                                $params['Encoding']='Byte'
+                            }
 
-                    # Notify client for file open.
-                    New-Event -SourceIdentifier PSESRemoteSessionOpenFile -EventArguments @($filePathName, $contentBytes) > $null
+                            $contentBytes = Get-Content @params
+
+                            # Notify client for file open.
+                            New-Event -SourceIdentifier PSESRemoteSessionOpenFile -EventArguments @($fileName, $contentBytes) > $null
+                        } else {
+                            $PSCmdlet.WriteError( (
+                                New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList @(
+                                    [System.Exception]'File already exists.'
+                                    $Null
+                                    [System.Management.Automation.ErrorCategory]::ResourceExists
+                                    $fileName ) ) )
+                        }
+                    }
                 }
             }
+
+            Set-Alias psedit Open-EditorFile -Scope Global
+            Export-ModuleMember -Function Open-EditorFile, New-EditorFile
         ";
 
         // This script is templated so that the '-Forward' parameter can be added
         // to the script when in non-local sessions
         private const string CreatePSEditFunctionScript = @"
             param (
-                [string] $PSEditFunction
+                [string] $PSEditModule
             )
 
             Register-EngineEvent -SourceIdentifier PSESRemoteSessionOpenFile -Forward
-
-            if ((Test-Path -Path 'function:\global:Open-EditorFile') -eq $false)
-            {{
-                Set-Item -Path 'function:\global:Open-EditorFile' -Value $PSEditFunction
-                Set-Alias psedit Open-EditorFile -Scope Global
-            }}
+            $psedit = New-Module -ScriptBlock ([Scriptblock]::Create($PSEditModule)) -Name PSEdit
+            $psedit.ExportedFunctions.Keys | ForEach-Object {
+                Set-Item -Path function:\global:$_ -Value $asdf.ExportedFunctions[$_].Definition -Force
+            }
```

Comment:
If we're going the module route, can we pipe this into `Import-Module` instead?

Reply:
@SeeminglyScience I tried `Import-Module` initially but it did not work. None of the cmdlets would show up... Could I have been missing a particular flag? I tried `-Force` and `-Scope`.

Reply:
@tylerl0706 try `-Global`.  I thought you only needed that when importing from a different session state, but I guess it's scope based. TIL